lib/sift/value_parser.rb

```diff
@@ -57,5 +57,28 @@ def boolean_value
         ActiveRecord::Type::Boolean.new.type_cast_from_user(value)
       end
     end
+
+    def normalized_value(value, type)
+      @value = value
+      if [:date, :time, :datetime].include?(type) && parse_as_range?
+        return normalized_date_range
+      end
+      value
+    end
+
+    def normalized_date_range
+      from_date_string, end_date_string = value.split("...")
+      return unless end_date_string
+
+      parsed_dates = [from_date_string, end_date_string].map do |date_string|
+        begin
+          DateTime.parse(date_string.to_s)
+        rescue StandardError
```

Comment:
would the value go through the [validator](https://github.com/procore/sift/pull/41/files#diff-8c2549f06a1b1eb9a9a29f07c99588c2R15) before this code is reached or after? I'm wondering if this rescue needs to be here

Reply:
validator is called on `filters_valid?` call https://github.com/procore/sift/blob/rename-to-sift/lib/sift.rb#L29

users of this gem can call `filtrate` https://github.com/procore/sift/blob/rename-to-sift/lib/sift.rb#L16 without `filters_valid?`